### DISCUSSION TITLE
How to do AND query with custom query ?

### DISCUSSION THREAD

**Discussion by Rirax** [strapi/strapi#7052] [Legacy Topics]
Hello,

 I try to create a custom query to perform an AND query but not working.

I have accomodation and amenities models.
Each accomodation have many amenities.

I would like to request the accommodations that have amenity 1 AND amenity 2.

For this I have created a custom controller : 
_/api/test/controllers/Test.js_

```
module.exports = {
  index: async ctx => {
    console.log(ctx.query);
    const result = await strapi
      .query('accomodation')
      .model.query(qb => {
        qb.where('amenities', [1, 2]);
      })
      .fetch();
  },
};
```

In dev environment I get t error:
Error: SQLITE_ERROR: no such column: amenities
Note that the commented code
`const result = await strapi.query('accomodation').find({ _limit: 10, amenities_in: [1, 2] });`
does return a result (but not possible to do AND query with this)

So what do I have to do to make the request work? And which operator to use for AND query ?
Bookshelf doc is not helping me :(

**Comment by derrickmehaffy** [comment]
Hi @Rirax with the release of the 3.1.x version we added some complex query options, you can take a look at this documentation for more information: https://strapi.io/documentation/v3.x/content-api/parameters.html#complex-queries

**Reply by Rirax** [reply]
hello @derrickmehaffy ! thank you for letting me know about a solution I'm really grateful ;) I'll look to migrate to 3.1
